docker-app/qfieldcloud/core/management/commands/extractstoragemetadata.py

```diff
@@ -0,0 +1,141 @@
+                acc["missing_value"].append(key)
+            return acc
+
+        reported = reduce(reducer, contents.items(), reporter)
+        reported["missing_key"] = [
+            key for key in s3_credentials_keys if key not in contents.keys()
+        ]
+
+        if any(reported.values()):
+            logger.error(f"Invalid config!: {reported}")
+            sys.exit(1)
+
+        return contents
+
+    @classmethod
+    def get_or_load(cls, path_to_file: str | None) -> S3ConfigObject:
+        """Get or create configuation for S3 storage."""
+        if not cls._config:
+            if not path_to_file:
+                logger.error(
+                    "You need to pass a valid path to your YAML configuration file"
+                )
+                sys.exit(1)
+            maybe_valid_config = cls.from_file(path_to_file)
+            valid_config = cls.validate(maybe_valid_config)
+            cls._config = S3ConfigObject(**valid_config)
+
+        return cls._config
+
+    @classmethod
+    def inject_to_settings(cls):
+        """To avoid passing s3 config as parameters"""
+        for key, value in cls._config._asdict().items():
+            setattr(settings, key, value)
+
+
+class Command(BaseCommand):
+    """Save metadata from S3 storage project files to disk."""
+
+    def add_arguments(self, parser: CommandParser):
+        parser.add_argument("-o", "--output", type=str, help="Name of output file")
+        parser.add_argument(
+            "-f",
+            "--config_file",
+            type=str,
+            help="Path to YAML file with your S3 credentials",
+        )
+
+    def handle(self, *args, **options):
+        if options.get("config_file"):
+            path = Path(options["config_file"])
+
+            if not path.is_file():
+                logger.error(f"This path does not exist or is not a file: {path}")
+                sys.exit(1)
+
+            S3Config.get_or_load(path)
+            S3Config.inject_to_settings()
+
+        output_name = options.get("output")
+
+        if output_name:
+            name_suffix = output_name.rsplit(".", 1)
+
+            if len(name_suffix) != 2 or name_suffix[1] != "csv":
+                logger.error("The name of the output file should end with '.csv'")
+                sys.exit(1)
+        else:
+            output_name = "s3_storage_files.csv"
+
+        interesting_fields = ["id", "key", "e_tag", "size", "create_time"]
+        bucket = utils.get_s3_bucket()
+
+        def gen_rows() -> Generator[list[str], None, None]:
+            for file in bucket.object_versions.all():
+                row = []
+
+                for field in interesting_fields:
+                    if hasattr(file, field):
+                        item = getattr(file, field)
+
+                        if not isinstance(item, str):
+                            item = str(item)
+
+                        item = item.strip('"')
+                        row.append(item)
```

Comment:
We do not need such level of abstraction with `interesting_fields`. You can directly get the keys the version object.